indexd/src/app_client.rs

```diff
@@ -96,6 +98,73 @@ pub struct ObjectsCursor {
     pub key: Hash256,
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
+pub struct GeoLocation {
+    pub latitude: f64,
+    pub longitude: f64,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
+pub enum HostSort {
+    Distance(GeoLocation),
+}
+
+#[derive(Debug, Clone, Default, PartialEq, Deserialize)]
+pub struct HostQuery {
+    pub sort: Option<HostSort>,
+    pub offset: Option<u64>,
+    pub limit: Option<u64>,
+    pub service_account: Option<bool>,
+    pub protocol: Option<Protocol>,
+    pub country: Option<String>,
+}
+
+impl HostQuery {
+    fn is_empty(&self) -> bool {
+        self.sort.is_none()
+            && self.offset.is_none()
+            && self.limit.is_none()
+            && self.service_account.is_none()
+            && self.protocol.is_none()
+            && self
+                .country
+                .as_ref()
+                .map(|country| country.is_empty())
+                .unwrap_or(true)
+    }
+}
+
+impl Serialize for HostQuery {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
```

Comment:
This seems like it might be hiding some issue. You are effectively implementing what seems to be the derived default implementation for `Serialize`. Apart from the location. If that is the case, the custom implementation of `Serialize` should be moved down as far as possible to the concrete type that needs it.

If the issue is related to `None` fields being serialized, you should be able to use `skip_serializing_if` to skip fields that are `None` rather than implementing this by hand.